htdocs/product/class/product.class.php

```diff
@@ -878,18 +878,19 @@ function update($id, $user, $notrigger=false, $action='update')
 	/**
 	 *  Delete a product from database (if not used)
 	 *
+	 * 	@param	    User	$user		Object user that ask to delete
 	 *	@param      int		$id         Product id (usage of this is deprecated, delete should be called without parameters on a fetched object)
 	 *  @param      int     $notrigger  Do not execute trigger
 	 * 	@return		int					< 0 if KO, 0 = Not possible, > 0 if OK
 	 */
-	function delete($id=0, $notrigger=0)
+	function delete($user, $id=0, $notrigger=0)
```

Comment:
This will break dolibarr. If you want to remove deprecated id parameter, you should replace it by the user object and ignore first parameter if int. Also better put this is separate PR.